src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostCodeActionsEndpointTest.cs

```diff
@@ -1159,6 +1159,98 @@ @using System
                     """)]);
     }
 
+    [FuseFact]
+    public async Task WrapAttributes()
+    {
+        await VerifyCodeActionAsync(
+            input: """
+                <div>
+                    <div [||]bar="Baz" Zip="Zap" checked @onclick="foo" Pop="Pap">
+                        <div></div>
+                    </div>
+                </div>
+                """,
+            expected: """
+                <div>
+                    <div bar="Baz"
+                         Zip="Zap"
+                         checked
+                         @onclick="foo"
+                         Pop="Pap">
+                        <div></div>
+                    </div>
+                </div>
+                """,
+            codeActionName: LanguageServerConstants.CodeActions.WrapAttributes);
+    }
+
+    [FuseFact]
+    public async Task WrapAttributes_Component()
+    {
+        await VerifyCodeActionAsync(
+            input: """
+                <div>
+                    <EditForm [||]bar="Baz" Zip="Zap" checked @onclick="foo" Pop="Pap" />
+                </div>
+                """,
+            expected: """
+                <div>
+                    <EditForm bar="Baz"
+                              Zip="Zap"
+                              checked
+                              @onclick="foo"
+                              Pop="Pap" />
+                </div>
+                """,
+            codeActionName: LanguageServerConstants.CodeActions.WrapAttributes);
+    }
+
+    [FuseFact]
+    public async Task WrapAttributes_Whitespace()
+    {
+        await VerifyCodeActionAsync(
+            input: """
+                <div>
+                    <Foo Bar="Baz"        Zip="Za[||]p"               Pop="Pap" />
+                </div>
+                """,
+            expected: """
+                <div>
+                    <Foo Bar="Baz"
+                         Zip="Zap"
+                         Pop="Pap" />
+                </div>
+                """,
+            codeActionName: LanguageServerConstants.CodeActions.WrapAttributes);
+    }
+
+    [FuseFact]
+    public async Task WrapAttributes_MultiLine()
+    {
+        await VerifyCodeActionAsync(
+            input: """
+                <div>
+                    <Foo Bar="Baz" Zip="Za[||]p"
+                         Pop="Pap" />
+                </div>
+                """,
+            expected: null,
```

Comment:
This seems a little inconsistent with the previous case. If we are in the attribute value and there is one more attribute after, we wrap both current and following attribute. But if we are in the attribute value and next attribute is on the next line, we do nothing? I would've expected us to wrap the current attribute to be more consistent with previous case.

Reply:
I took the approach of only offering wrapping if everything was on one line. If they're on multiple lines the code action doesn't offer itself, but formatting would take care of moving things on the line if necessary.

Certainly could expand it later to handle more cases.